hatchery/transitgateway.go

```diff
@@ -0,0 +1,579 @@
+package hatchery
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ram"
+)
+
+func setupTransitGateway(username string) error {
+	_, err := createTransitGateway(username)
+	if err != nil {
+		return err
+	}
+	Config.Logger.Printf("Setting up remote account ")
+	err = setupRemoteAccount(username, false)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func teardownTransitGateway(username string) error {
+	err := setupRemoteAccount(username, true)
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+func describeMainNetwork(vpcid string, svc *ec2.EC2) (*NetworkInfo, error) {
```

Comment:
what is the difference between "main network" and "workspace network" in ec2.go?

Reply:
The network in the main AWS account, and the network in the workspace aws account.